deepfence_server/reporters/filter.go

```diff
@@ -71,7 +71,20 @@ func containsFilter2CypherConditions(cypherNodeName string, filter ContainsFilte
 		reverseOperator = " NOT "
 	}
 	for k, vs := range filter.FieldsValues {
-		if k == "node_type" && nodeTypeCondition[k] {
+		acceptedNodeTypesFound := true
+		if k == "node_type" {
+			for _, v := range vs {
+				if !nodeLabelsAvailableForNodeType[v.(string)] {
+					acceptedNodeTypesFound = false
+					break
+				}
```

Comment:
If we use previous, we can simplify with: 
```suggestion
				_, acceptedNodeTypesFound = nodeLabelsAvailableForNodeType[v.(string)] {
```